testament/testament.nim

```diff
@@ -847,82 +840,73 @@ proc needsCodegenCheck(run: TestRun): bool =
 proc codegenCheck(
     test: TTest,
     target: TTarget,
-    spec: TSpec,
-    expectedMsg: var string,
-    given: var CompilerOutput
-  ) =
+    spec: TSpec
+  ): TestResult =
   ## Check for any codegen mismatches in file generated from `test` run.
   ## Only file that was immediately generated is tested.
```

Comment:
Just remembered how buggy the pre-existing code is wrt reporting all the various issues, code size errors can clobber codegen errors that were detected and only the last check is retained on mismatch.

Soft suggestion: might be worth leaving a `TODO` for someone else or future PR in this series.

```suggestion
  ## Only file that was immediately generated is tested.

  # xxx: only reports last failure, an iterator or early exit might be better?
```